pkg/controller/deployment/deployment_helper.go

```diff
@@ -76,6 +76,74 @@ func parseArgMap(argMap map[string]string, args []string) {
 	}
 }
 
+// mergeContainerResources merges source container resources with that
+// provided as override resources.
+func mergeContainerResources(sourceResources corev1.ResourceRequirements, overrideResources corev1.ResourceRequirements) corev1.ResourceRequirements {
```

Comment:
Is it possible to make use of https://github.com/kubernetes/kubernetes/blob/7f6f68fdabc4df88cfea2dcf9a19b2b830f1e647/pkg/api/v1/resource/helpers.go#L377?

Reply:
+1

Reply:
The func signature in this `MergeContainerResourceLimits` fn uses `v1.Container` type and two things to note:
- it doesn't set `hugepage` related resources
- it doesn't set the overriden limit in case the limit was already set in the container (i.e. if we use this and in the future we end up adding resource *limits* to our operator deployment (which is currently not present) we have to change this implementation)

Reply:
signature change shouldn't matter to us, but anyways we can't reuse. 

> it doesn't set hugepage related resources
We do not want to support hugepages either since this will involve overriding volumes and volumemounts. So until someone files a bug/rfe. 

> it doesn't set the overriden limit in case the limit was already set in the container (i.e. if we use this and in the future we end up adding resource limits to our operand deployment (which is currently not present) we have to change this implementation)
Yeah, this might be an issue.